### PR TITLE
Editing Groups

### PR DESCRIPTION
**Pull request by pnor**
## Overview

Created Edit Group Screen / Refactored Edit Interest to also work as groups, finishing up the profile editing screens

## Changes Made

- Refactor EditInterestsVC to be able to use groups as well
- Add optional search bar to headers
- Change section struct to class for more flexibility on filtering search results
- Change labels and names where appropriate
## Test Coverage

Ran in simulator

## Next Steps (delete if not applicable)

Integrate with backend

## Related PRs or Issues (delete if not applicable)

#38 

## Screenshots (delete if not applicable)

<details>

  <summary>Group</summary>

group  

</details>

<details>

  <summary>Interest (the title is wrong, but is fixed from commit "Fix title")</summary>

inter
</details>

### CODE REVIEW

File: CoffeeChat/SceneDelegate.swift [hunk 1]
```diff
@@ -40,12 +40,16 @@ class SceneDelegate: UIResponder, UIWindowSceneDelegate {
...
+            // let navigationController = UINavigationController(rootViewController: rootVC)
+            let testvc = EditInterestViewController()
+            let navigationcontroller = UINavigationController(rootViewController: testvc)
+            window.rootViewController = navigationcontroller
         } else {
             // Ask user to sign in if they have not signed in before.
-            let navigationController = UINavigationController(rootViewController: LoginViewController())
-            window.rootViewController = navigationController
+            //let navigationController = UINavigationController(rootViewController: LoginViewController())
+            let testvc = EditInterestViewController()
```
Comment (kevinchan159) [inline CoffeeChat/SceneDelegate.swift:50]: prob want to remove the testvc before merging?

File: CoffeeChat/Controller/EditInterestViewController.swift [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+
+    func removeItem(named name: String) -> ItemType? {
+        if let loc = items.firstIndex(where: { $0.getName() == name }) {
+            let removed = items.remove(at: loc)
+            items.sort(by: { $0.getName() < $1.getName() })
+            refilter()
+            return removed
+        }
+        return nil
+
```
Comment (haiyingweng) [inline CoffeeChat/Controller/EditInterestViewController.swift:64]: remove line

File: CoffeeChat/Controller/EditInterestViewController.swift [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+}
+
+// MARK: - UIViewController
+class EditInterestViewController: UIViewController {
+
+    // MARK: - Private View Vars
+    private let tableView = UITableView(frame: .zero, style: .grouped)
+    private let backButton = UIButton()
+    private let saveBarButtonItem = UIBarButtonItem()
```
Comment (haiyingweng) [inline CoffeeChat/Controller/EditInterestViewController.swift:83]: alphabetize?

File: CoffeeChat/Controller/EditInterestViewController.swift [hunk 4]
```diff
@@ -0,0 +1,533 @@
...
+        tableView.reloadData()
+    }
+
+}
+
+// MARK: - UITableViewDataSource
+extension EditInterestViewController: UITableViewDataSource {
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return section == 0
```
Comment (haiyingweng) [inline CoffeeChat/Controller/EditInterestViewController.swift:262]: switch on section type?

File: CoffeeChat/Controller/EditInterestViewController.swift [hunk 5]
```diff
@@ -0,0 +1,533 @@
...
+            cell.changeColor(isSelected: section.type == .yours)
+            cell.selectionChangesAppearence(false)
+            return cell
+        }
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = EditHeaderView()
+
+        if section == 0 { // Upper Section
```
Comment (haiyingweng) [inline CoffeeChat/Controller/EditInterestViewController.swift:291]: same with switch on section type

File: CoffeeChat/Controller/EditInterestViewController.swift [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+    func addItem(_ item: ItemType) {
+        items.append(item)
+        items.sort(by: sortStrategy)
+        refilter()
+    }
+
+    func removeItem(named name: String) -> ItemType? {
+        if let loc = items.firstIndex(where: { $0.getName() == name }) {
+            let removed = items.remove(at: loc)
+            items.sort(by: { $0.getName() < $1.getName() })
```
Comment (lucyxubroad) [inline CoffeeChat/Controller/EditInterestViewController.swift:59]: can you just use ```sortStrategy``` here?

**Comment by lucyxubroad** [comment]
Also random but let's change the title of the PR. I got yelled at by Young about this some time ago heh

**Pull request merged**